mozjs/tests/external_string.rs

```diff
@@ -36,21 +36,45 @@ fn external_string() {
         ));
         let _ac = JSAutoRealm::new(context, global.get());
 
-        let latin1_base = "test latin-1";
-        let latin1_boxed = latin1_base.as_bytes().to_vec().into_boxed_slice();
-        let latin1_chars = Box::into_raw(latin1_boxed).cast::<u8>();
-
-        let callbacks = CreateJSExternalStringCallbacks(
-            &EXTERNAL_STRING_CALLBACKS_TRAPS,
-            latin1_base.len() as *mut c_void,
+        test_latin1_string(context, "test latin1");
+        test_latin1_string(context, "abcdefghijklmnop"); // exactly 16 bytes
+        test_latin1_string(context, "abcdefghijklmnopq"); // 17 bytes
+        test_latin1_string(context, "abcdefghijklmno"); // 15 bytes
+        test_latin1_string(context, "abcdefghijklmnopqrstuvwxyzabcdef"); //32 bytes
+        test_latin1_string(context, "abcdefghijklmnopqrstuvwxyzabcde"); //31 bytes
+        test_latin1_string(context, "abcdefghijklmnopqrstuvwxyzabcdefg"); //33 bytes
```

Comment:
I suspect this will fail either clippy or rustfmt checks otherwise: 

```suggestion
        test_latin1_string(context, "abcdefghijklmnopqrstuvwxyzabcdef"); // 32 bytes
        test_latin1_string(context, "abcdefghijklmnopqrstuvwxyzabcde"); // 31 bytes
        test_latin1_string(context, "abcdefghijklmnopqrstuvwxyzabcdefg"); // 33 bytes
```